synapse/http/server.py

```diff
@@ -270,12 +266,11 @@ def register_paths(self, method, path_patterns, callback):
     def render(self, request):
         """ This gets called by twisted every time someone sends us a request.
         """
-        self._async_render(request)
+        defer.ensureDeferred(self._async_render(request))
```

Comment:
Presumably we need to wrap this in a deferred to make it run on the reactor?

Reply:
Yep, the coroutine itself does not do anything until it's either awaited or wrapped in a Deferred.